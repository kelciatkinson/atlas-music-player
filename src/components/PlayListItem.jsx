export default function PlayListItem({title, artist, length}) {
  return <div className="space-y-1">
    <div className="group flex justify-between hover:bg-(--color-accent) transition duration-300 ease-in-out dark:bg-(--color-dark-accent) px-0.25 pb-0.5 rounded-lg">
      <div>
        <p className="font-medium text-(--color-text) group-hover:text-(--color-text) transition duration-300 ease-in-out dark:text-(--color-text)">Painted in Blue</p>
        <p className="font-medium text-sm text-(--color-secondary) group-hover:text-(--color-text) transition duration-300 ease-in-out dark:text-(--color-dark-secondary) pb-1">Soul Canvas</p>
      </div>
      <p className="self-center font-medium text-sm text-(--color-secondary) group-hover:text-(--color-text) transition duration-300 ease-in-out dark:text-(--color-text)">5:55</p>
    </div>
  </div>
}
