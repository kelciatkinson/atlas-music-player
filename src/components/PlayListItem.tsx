import React from "react";

interface PlayListItemProps {
  title: string;
  artist: string;
  duration: number;
  isActive?: boolean;
  onClick?: () => void;
}
export default function PlayListItem({ title, artist, duration, isActive, onClick }: PlayListItemProps) {
  // duration is in seconds
  const minutes = Math.floor(duration / 60);
  const seconds = Math.floor(duration) % 60;
  let stringMinutes;
  let stringSeconds;
  console.log("Minutes: ", minutes);

  if (minutes < 10) {
    stringMinutes = minutes.toString();
    stringMinutes = `0${stringMinutes}`;
  } else {
    stringMinutes = minutes.toString();
  }
  if (seconds < 10) {
    stringSeconds = seconds.toString();
    stringSeconds = `0${stringSeconds}`;
  } else {
    stringSeconds = seconds.toString();
  }

  console.log(`${title} isActive`, isActive);
  return <div className="space-y-1">
    <div onClick={onClick} className={`group flex justify-between ${isActive ? 'bg-orange-200' : ''} hover:opacity-50 transition duration-300 ease-in-out px-0.25 pb-0.5 rounded-lg`}>
      <div>
        <p className="font-medium text-(--color-text) group-hover:text-(--color-text) transition duration-300 ease-in-out">{title}</p>
        <p className="font-medium text-sm text-(--color-secondary) group-hover:text-(--color-text) transition duration-300 ease-in-out pb-1">{artist}</p>
      </div>
      <p className="self-center font-medium text-sm text-(--color-secondary) group-hover:text-(--color-text) transition duration-300 ease-in-out">{stringMinutes}:{stringSeconds}</p>
    </div>
  </div>
}
