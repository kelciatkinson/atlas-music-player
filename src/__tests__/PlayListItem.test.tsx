import { render } from "@testing-library/react";
import { expect, test, describe } from "vitest";
import PlayListItem from "../components/PlayListItem";
import React from "react";


describe("PlayListItem Snapshot Tests", () => {
  test("renders with basic props", () => {
    const { container } = render(
      <PlayListItem 
        title="Song Title"
        artist="Artist Name"
        duration={180}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders with inactive state", () => {
    const { container } = render(
      <PlayListItem 
        title="Inactive Song"
        artist="Another Artist"
        duration={300}
        isActive={false}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders with long duration (hours)", () => {
    const { container } = render(
      <PlayListItem 
        title="Long Song"
        artist="Epic Artist"
        duration={3665}
      />
    );
    expect(container).toMatchSnapshot();
  });
});