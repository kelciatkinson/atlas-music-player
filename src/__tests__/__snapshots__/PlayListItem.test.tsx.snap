// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem Snapshot Tests > renders with basic props 1`] = `
<div>
  <div
    class="space-y-1"
  >
    <div
      class="group flex justify-between  hover:opacity-50 transition duration-300 ease-in-out px-0.25 pb-0.5 rounded-lg"
    >
      <div>
        <p
          class="font-medium text-(--color-text) group-hover:text-(--color-text) transition duration-300 ease-in-out"
        >
          Song Title
        </p>
        <p
          class="font-medium text-sm text-(--color-secondary) group-hover:text-(--color-text) transition duration-300 ease-in-out pb-1"
        >
          Artist Name
        </p>
      </div>
      <p
        class="self-center font-medium text-sm text-(--color-secondary) group-hover:text-(--color-text) transition duration-300 ease-in-out"
      >
        0
        :
        3
      </p>
    </div>
  </div>
</div>
`;

exports[`PlayListItem Snapshot Tests > renders with inactive state 1`] = `
<div>
  <div
    class="space-y-1"
  >
    <div
      class="group flex justify-between  hover:opacity-50 transition duration-300 ease-in-out px-0.25 pb-0.5 rounded-lg"
    >
      <div>
        <p
          class="font-medium text-(--color-text) group-hover:text-(--color-text) transition duration-300 ease-in-out"
        >
          Inactive Song
        </p>
        <p
          class="font-medium text-sm text-(--color-secondary) group-hover:text-(--color-text) transition duration-300 ease-in-out pb-1"
        >
          Another Artist
        </p>
      </div>
      <p
        class="self-center font-medium text-sm text-(--color-secondary) group-hover:text-(--color-text) transition duration-300 ease-in-out"
      >
        0
        :
        5
      </p>
    </div>
  </div>
</div>
`;

exports[`PlayListItem Snapshot Tests > renders with long duration (hours) 1`] = `
<div>
  <div
    class="space-y-1"
  >
    <div
      class="group flex justify-between  hover:opacity-50 transition duration-300 ease-in-out px-0.25 pb-0.5 rounded-lg"
    >
      <div>
        <p
          class="font-medium text-(--color-text) group-hover:text-(--color-text) transition duration-300 ease-in-out"
        >
          Long Song
        </p>
        <p
          class="font-medium text-sm text-(--color-secondary) group-hover:text-(--color-text) transition duration-300 ease-in-out pb-1"
        >
          Epic Artist
        </p>
      </div>
      <p
        class="self-center font-medium text-sm text-(--color-secondary) group-hover:text-(--color-text) transition duration-300 ease-in-out"
      >
        1
        :
        1
      </p>
    </div>
  </div>
</div>
`;
